server/controllers/documents.js

```diff
@@ -0,0 +1,141 @@
+const Document = require('../models').documents;
+const User = require('../models/').users;
+
+module.exports = {
+  create(req, res) {
+    return Document
+      .create({
+        title: req.body.title,
+        content: req.body.content,
+        access: req.body.access,
+        userId: req.body.userId,
+      })
+      .then(doc => res.status(201).send(doc))
+      .catch(error => res.status(400).send(error));
+  },
+
+  list(req, res) {
+    return Document
+      .findAll()
+      .then(doc => {
+         if ( doc.length < 1 ) {
+          return res.status(404).send({
+            message: 'There are no documents created.',
+          });
+        }
+        return res.status(200).send(doc);
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  retrieve(req, res) {
+    return Document
+      .findById(req.params.documentId)
+      .then(doc => {
+        if (!doc) {
+          return res.status(404).send({
+            message: 'Document Not Found',
+          });
+        }
+        return res.status(200).send(doc);
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  getUserDocuments(req, res) {
+    return Document
+      .findAll({
+        where: { userId: req.params.userId }
+      })
+      .then(doc => {
+        if (!doc || doc.length < 1) {
+          return res.status(404).send({
+            message: 'That user either does not exist or has no documents',
+          })
+        }
+        return res.status(200).send(doc)
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  update(req, res) {
+    return Document
+      .findById(req.params.documentId)
+      .then(doc => {
+        if (doc == []) {
```

Comment:
**Code Quality - Nitpick**

- `Model.findById(id)` will return a promise which on resolution receives an instance of the model that matches the id passed and not an array. I'd rather have `if (!doc) {...}`